Sources/TuistGenerator/Generator/SchemeDescriptorsGenerator.swift

```diff
@@ -440,6 +440,22 @@ final class SchemeDescriptorsGenerator: SchemeDescriptorsGenerating {
             }
         }
 
+        if let expandVariableFromTarget = scheme.runAction?.expandVariableFromTarget {
+            guard let graphTarget = graphTraverser.target(
+                path: expandVariableFromTarget.projectPath,
+                name: expandVariableFromTarget.name
+            ) else { return nil }
+
+            guard let buildableReference = try createBuildableReference(
+                graphTarget: graphTarget,
+                graphTraverser: graphTraverser,
+                rootPath: rootPath,
+                generatedProjects: generatedProjects
+            ) else { return nil }
+
+            macroExpansion = buildableReference
```

Comment:
It seems Xcode has a small optimisation where if there's a runnable / executable specified it by default be the target variables are expanded from even if `macroExpansion` isn't specified.

If the runnable target is explicitly specified and assigned to `macroExpansion`, viewing and closing the scheme editor would remove the `macroExpansion` from the `.xcscheme` on disk 😅

Potentially can check if `buildableProductRunnable?.buildableReference` is the same and skip assignment of `macroExpansion` to ensure a stable scheme (with a comment to that extent).